hnn_core/viz.py

```diff
@@ -1222,3 +1235,365 @@ def plot_laminar_csd(times, data, contact_labels, ax=None, colorbar=True,
     plt_show(show)
 
     return ax.get_figure()
+
+
+class NetworkPlotter:
+    """Helper class to visualize full morphology of HNN model.
+
+    Parameters
+    ----------
+    net : Instance of Network object
+        The Network object
+    ax : instance of matplotlib Axes3D | None
+        An axis object from matplotlib. If None,
+        a new figure is created.
+    vmin : int | float
+        Lower limit of colormap for plotting voltage
+        Default: -100 mV
+    vmax : int | float
+        Upper limit of colormap for plotting voltage
+        Default: 50 mV
+    bg_color : str
+        Background color of ax. Default: 'black'
+    colorbar : bool
+        If True (default), adjust figure to include colorbar.
+    voltage_colormap : str
+        Colormap used for plotting voltages
+        Default: 'viridis'
+    elev : int | float
+        Elevation 3D plot viewpoint, default: 10
+    azim : int | float
+        Azimuth of 3D plot view point, default: 20
+    xlim : tuple of int | tuple of float
+        x limits of plot window. Default (-200, 3100)
+    ylim : tuple of int | tuple of float
+        y limits of plot window. Default (-200, 3100)
+    zlim : tuple of int | tuple of float
+        z limits of plot window. Default (-300, 2200)
+    trial_idx : int
+        Index of simulation trial plotted. Default: 0
+    time_idx : int
+        Index of time point plotted. Default: 0
+    """
+    def __init__(self, net, ax=None, vmin=-100, vmax=50, bg_color='black',
+                 colorbar=True, voltage_colormap='viridis', elev=10, azim=-500,
+                 xlim=(-200, 3100), ylim=(-200, 3100), zlim=(-300, 2200),
+                 trial_idx=0, time_idx=0):
+        import matplotlib.pyplot as plt
+        from matplotlib import colormaps
+        self.net = net
+
+        # Check if network simulated
+        if net.cell_response is not None:
+            self.times = net.cell_response.times
+
+            # Check if voltage recorded
+            if net._params['record_vsec'] == 'all':
+                self._vsec_recorded = True
+            else:
+                self._vsec_recorded = False
+        else:
+            self._vsec_recorded = False
+            self.times = None
+
+        _validate_type(vmin, (int, float), 'vmin')
```

Comment:
I would suggest moving all the type validation to the top of the init for both logic and readability. I'm assuming that if any of the validations fail an error is being raised and you won't continue with the init?